packages/template-retail-react-app/app/components/item-variant/item-price.jsx

```diff
@@ -43,74 +40,66 @@ PricePerItem.propTypes = {
  */
 const ItemPrice = ({currency, align = 'right', baseDirection = 'column', ...props}) => {
     const variant = useItemVariant()
-    const {data: basket} = useCurrentBasket()
-    const {currency: activeCurrency} = useCurrency()
-    const intl = useIntl()
-
-    const {price, basePrice, priceAfterItemDiscount} = variant
-    const isProductASet = variant?.type?.set
-
-    const displayPrice =
-        typeof priceAfterItemDiscount === 'number' ? Math.min(price, priceAfterItemDiscount) : price
-
-    const hasDiscount = displayPrice !== price
-
+    const {price, priceAfterItemDiscount} = variant
+    const isASet = variant?.type?.set
+    const isMaster = variant?.type?.master
+    let priceData
+    // When variant has basket pricing, we should prioritize these prices for displaying
+    // since they may have take promotion/discount into account
+    // NOTE: try NOT to re-calculate these values since basket-level discount is complicated
+    if (variant?.itemId) {
+        priceData = {
+            currentPrice: priceAfterItemDiscount,
+            listPrice: undefined,
+            pricePerUnit: variant?.pricePerUnit,
+            isASet,
+            isMaster,
+            isRange: isASet || isMaster || false,
+            isOnSale: price > priceAfterItemDiscount
+        }
+    } else {
+        // for wishlist page we extract price info from variant/product obj
+        priceData = getPriceData(variant)
+    }
+    const isDesktop = useBreakpointValue({base: false, lg: true})
+    const isOnSale = price > priceAfterItemDiscount || priceData?.isOnSale
     return (
         <Stack
             textAlign={align}
-            direction={hasDiscount ? 'column' : {base: baseDirection, lg: 'row'}}
+            direction="column"
```

Comment:
The `baseDirection` prop is not used anywhere in this component. How about using it here?

```suggestion
            direction={baseDirection}
```